server.c

```diff
@@ -2402,13 +2418,25 @@ server_main(struct nsd *nsd)
 					restart_child_servers(nsd, server_region, netio,
 						&nsd->xfrd_listener->fd);
 				} else if (child_pid == reload_pid) {
-					sig_atomic_t cmd = NSD_RELOAD_DONE;
+					const char *fmt;
+					sig_atomic_t cmd = NSD_RELOAD_FAILED;
 					pid_t mypid;
-					log_msg(LOG_WARNING,
-					       "Reload process %d failed with status %d, continuing with old database",
-					       (int) child_pid, status);
+					if(WIFEXITED(status)) {
+						fmt = "Reload process %d "
+						      "exited with status %d, "
+						      "continuing with old "
+						      "database";
+						status = WEXITSTATUS(status);
+					} else {
+						fmt = "Reload process %d "
+						      "exited abnormally, "
+						      "continuing with old "
+						      "database";
+					}
+					log_msg(LOG_WARNING, fmt, child_pid, status);
```

Comment:
Could you put the log_msg twice, for the two fmt options. Above in the if statement.  Then also the number of arguments is correct, and checked by the compiler.  I think that is nicer, not that it is wrong now.